tools/zetasql_helper/zetasql_helper/local_service/local_service.cc

```diff
@@ -33,5 +36,51 @@ absl::Status ZetaSqlHelperLocalServiceImpl::Tokenize(
   return absl::OkStatus();
 }
 
+absl::Status ZetaSqlHelperLocalServiceImpl::ExtractFunctionRange(
+    const ExtractFunctionRangeRequest *request,
+    ExtractFunctionRangeResponse *response) {
+
+  std::unique_ptr<::bigquery::utils::zetasql_helper::FunctionRange> output;
+  ZETASQL_RETURN_IF_ERROR(
+      ::bigquery::utils::zetasql_helper::ExtractFunctionRange(
+          request->query(), request->line_number(), request->column_number(), &output
+      ));
+
+  auto status_or_function_range_proto = output->ToProto();
+  ZETASQL_RETURN_IF_ERROR(status_or_function_range_proto.status());
+
+  // Convert the proto to heap value and hand over the ownership to the response.
+  auto function_range_proto = new ::bigquery::utils::zetasql_helper::FunctionRangeProto(
+      status_or_function_range_proto.value()
```

Comment:
you can use unique_ptr and std::move?

Reply:
I think this is how proto specifies the ownership - assign a pointer to its field. The child pointer will be released automatically when the owner is released. We don't need the unique_ptr here.